fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/indexes/StandardIndexMaintainer.java

```diff
@@ -669,4 +705,27 @@ public <M extends Message> List<IndexEntry> evaluateIndex(@Nonnull FDBRecord<M>
 
         return indexKeys.stream().map(key -> new IndexEntry(state.index, key)).collect(Collectors.toList());
     }
+
+    /**
+     * Create the list of index key-references that would allow the DB to prefetch the records pointed to by the index entries.
+     * The commonPrimaryKey is the representation of the primary key for the referenced records.
+     * The index structure is: [P1...Pn, I1...In, K1...Kn] where Px are the prefix elements, Ix are the index fields and Kx are the primary keys of the indexed record.
+     * Since {@link Index#trimPrimaryKey(List)} removes redundant key entries, we need to construct the list of locations of the primary key
+     * elements by using the keyLocations (if there are any), followed by the remaining key elements.
+     * @param recordSubspace the Record subspace (the prefix for the record entries)
+     * @param commonPrimaryKey the metadata of the primary key (used to construct the PK locations in teh dereferenced record)
+     * @return A Tuple representing the Mapper structure required by the FDB getMappedRange call
+     */
+    @Nonnull
+    private Tuple createRemoteFetchMapper(@Nonnull Subspace recordSubspace, @Nonnull KeyExpression commonPrimaryKey) {
```

Comment:
I suspect that as this feature grows, we'll want this in a separate helper class. But it's not burdensome yet.

Reply:
Agree. Although one possible extension vector is for different index maintainers to each have their own mapping utilities.